### PR TITLE
Pc spike try aggregate query

### PR DESCRIPTION
**Pull request by pconrad**
In this draft PR, we show an example of how to make an aggregate query on the MongoDB database.

We add a route `/api/public/statistics/courseCount` that returns JSON showing, for each quarter and deptCode, the number of courses in that department.

### CODE REVIEW

File: src/test/java/edu/ucsb/courses/documents/statistics/QuarterDeptTests.java [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+    }
+
+    @Test
+    public void test_equalsAnother() throws Exception {
+        QuarterDept qd1 = new QuarterDept("20204","CMPSC",0);
+        QuarterDept qd2 = new QuarterDept("20204","CMPSC",0);
+        qd2.setDeptCode("CMPSC"); 
+        qd2.setQuarter("20204");
+        assertEquals(qd1, qd2);
+    }
```
Comment (btk5h) [inline src/test/java/edu/ucsb/courses/documents/statistics/QuarterDeptTests.java:52]: What is this testing? Wouldn't the setters do nothing here?

File: src/main/java/edu/ucsb/courses/documents/statistics/QuarterDept.java [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+        EqualsBuilder builder = new EqualsBuilder();
+        builder.append(quarter, qd.getQuarter()).append(deptCode, qd.getDeptCode());
+        return builder.build();
+
+    }
+
+    @Override
+    public int hashCode() {
+        return Objects.hash(quarter, deptCode);
+    }
```
[the remark this answers is not in the document]
Comment (phtcon) [reply src/main/java/edu/ucsb/courses/documents/statistics/QuarterDept.java:77]: Yes

File: src/test/java/edu/ucsb/courses/documents/statistics/QuarterDeptTests.java (hunk 1 above)
Comment (phtcon) [reply src/test/java/edu/ucsb/courses/documents/statistics/QuarterDeptTests.java:52]: That's left over from a refactoring... good catch. I'll remove them.

Comment (pconrad) [reply src/test/java/edu/ucsb/courses/documents/statistics/QuarterDeptTests.java]: Fixed

File: src/main/java/edu/ucsb/courses/documents/statistics/QuarterDept.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (pconrad) [reply src/main/java/edu/ucsb/courses/documents/statistics/QuarterDept.java:77]: We should only compare on the items that identify the object.

**Pull request merged**